lib/manageiq/api/client/acts_as_ar_query.rb

```diff
@@ -0,0 +1,175 @@
+module ManageIQ
+  module API
+    class Client
+      class ActsAsArQuery
+        include Enumerable
+        attr_accessor :klass, :options
+
+        SORT_ASC = "asc".freeze
+        SORT_DESC = "desc".freeze
+
+        def initialize(klass, opts = nil)
+          @klass = klass
+          @options = opts || {}
+        end
+
+        def where(*val)
+          conditional(:where, val)
+        end
+
+        def not(*val)
+          conditional(:not, val)
+        end
+
+        def order(*val)
+          options[:order] ||= {}
+          val.flatten.each do |sval|
+            case sval
+            when String
+              sval.split(",").each do |value|
+                sort_attr, sort_order = value.split(" ")
+                options[:order][sort_attr.to_s] = sort_order || SORT_ASC
+              end
+            when Hash
+              sval.each do |sort_attr, sort_order|
+                options[:order][sort_attr.to_s] = sort_order
+              end
+            when Symbol
+              options[:order][sval.to_s] = SORT_ASC
+            end
+          end
+          self
+        end
+
+        def reorder(*val)
+          order(val)
+        end
+
+        def reverse_order
+          if options[:order].present?
+            options[:order] = options[:order].keys.each_with_object({}) { |key, hash| hash[key.to_s] = SORT_DESC }
+          end
+          self
+        end
+
+        def except(*val)
+          val.flatten.each do |key|
+            options.delete(key)
+          end
+          self
+        end
+
+        # similar to except, difference being this persists across merges.
+        def unscope(*val)
+          val.flatten.each do |key|
+            options[key] = nil
+          end
+          self
+        end
+
+        def only(*val)
+          options.slice!(val)
+          self
+        end
+
+        def offset(val)
+          assign_arg :offset, val
+        end
+
+        def limit(val)
+          assign_arg :limit, val
+        end
+
+        def select(*args)
```

Comment:
FYI: active record supports both the array select and the sql column select. Not sure if we want to do something like this.

Think it is something like

```ruby
def select(*args, &block)
  if block_given?
    # user is specifying an enumerator. execute query and run the select
    to_a.select(&block)
  else
    # sql select clause, carry on
    append_hash_arg :select, *args
  end
end
```